llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -4846,6 +4846,26 @@ static MachineBasicBlock *emitIndirectDst(MachineInstr &MI,
   return LoopBB;
 }
 
+static uint32_t getInitialValueForWaveReduction(unsigned Opc){
+  switch(Opc){
+      case AMDGPU::S_MIN_U32:
+        return std::numeric_limits<uint32_t>::max();
+      case AMDGPU::S_MIN_I32:
+        return std::numeric_limits<int32_t>::max();
+      case AMDGPU::S_MAX_U32:
+        return 0;
+      case AMDGPU::S_MAX_I32:
+        return std::numeric_limits<int32_t>::min();
+      case AMDGPU::S_ADD_I32:
+      case AMDGPU::S_SUB_I32:
+      case AMDGPU::S_OR_B32:
+      case AMDGPU::S_XOR_B32:
+        return 0x00000000;
```

Comment:
Literal values are not consistent form here (0, 0x00000000, 0xFFFFFFFF). Can we not query this using std::numeric_limits here also?